File: DSA/strings/rotateString.java
class Solution {
    public boolean rotateString(String s, String goal) {
        if(s.length()!=goal.length()) return false;
        String doubledStr = s + s;
        return doubledStr.contains(goal);
    }
}

public class rotateString {
    public static void main(String[] args) {
        Solution sol = new Solution();
        String s = "abcde";
        String goal = "cdeab";
        System.out.println(sol.rotateString(s, goal)); // true

        s = "abcde";
        goal = "abced";
        System.out.println(sol.rotateString(s, goal)); // false
    }
}

File: DSA/strings/RotateString.cpp
class Solution{	
	public:
		bool rotateString(string& s,string& goal){
			if(s.size()!=goal.size()) return false;
            string doubledstr = s + s;
            return doubledstr.find(goal) != string::npos;
		}
};

#include <iostream>
#include <string>
using namespace std;

int main() {
    string s, goal;
    cout << "Enter original string: ";
    cin >> s;
    cout << "Enter goal string: ";
    cin >> goal;
    Solution sol;
    if (sol.rotateString(s, goal)) {
        cout << "True" << endl;
    } else {
        cout << "False" << endl;
    }
    return 0;
}